src/components/pages/ApartmentEditPage.jsx

```diff
@@ -56,46 +80,71 @@ const ApartmentEditPage = ({
 
         const getInitialValues = memoize(ApartmentFormFields.getInitialValues);
 
-        if (editPath !== null && isRouteApartmentPresent) {
+        if (editPath !== null && this.isRouteApartmentPresent()) {
             const apartmentId = editPath.params.apartmentId;
             const apartment = apartments.find(
                 (item) => item.id === apartmentId
             );
-            initialValues = getInitialValues(apartment);
+            return getInitialValues(apartment);
         }
 
         if (createPath !== null) {
             const apartment = new Apartment();
-            initialValues = getInitialValues(apartment);
+            return getInitialValues(apartment);
         }
     };
 
-    useEffect(updateInitialValuesFromLocation, [
-        location,
-        isRouteApartmentPresent,
-    ]);
-
-    const { t } = useTranslation();
-
-    return (
-        <PageCard header={t("apartment.edit")}>
-            <InProgress inProgress={apartmentsInProgress} />
-            <PageErrorMessage isError={apartmentsErrorMessage}>
-                {apartmentsErrorMessage}
-            </PageErrorMessage>
-            {showApartmentNotFoundVisible && (
-                <Alert color="warning">{t("apartment.not_found")}</Alert>
-            )}
-            {showEditForm && (
-                <ApartmentForm
-                    apartmentInProgress={"FIXME_PUT_PROGRESS_TYPE_HERE"}
-                    initialValues={initialValues}
-                    onSubmit={onSubmit}
-                />
-            )}
-            <PageNavigationApartmentList />
-        </PageCard>
-    );
-};
-
-export default withApartments(ApartmentEditPage);
+    hasApartments = () =>
+        this.props.apartmentsSuccess && this.props.apartments.length > 0;
+
+    isRouteApartmentPresent = () =>
+        this.props.apartments.some(
+            (apartment) => apartment.id === this.props.match.params.apartmentId
+        );
+
+    renderForm = () => {
+        if (this.hasApartments()) {
```

Comment:
Just a minor neatpick / code style: if you have a single if condition with a big chunk of code executed in it, and then a short alternative (like "return null" here), just negate the condition and make it like `if (!condition) { return ... } ", i.e. exit early. This reduces the number of indentations required in order to actually keep and allows to focus on the code more easly. But that's just my opinion / suggestion, I don't feel strongly tied to it :)